aikau/src/main/resources/alfresco/prototyping/Preview.js

```diff
@@ -362,56 +401,50 @@ define(["dojo/_base/declare",
       updatePage: function alfresco_prototyping_Preview__updatePage(response, originalRequestConfig) {
          // Iterate over the CSS map and append a new <link> element into the <head> element to ensure that all the
          // widgets CSS dependencies are loaded... 
-         for (var media in response.cssMap)
-         {
-            if (response.cssMap.hasOwnProperty(media))
-            {
-               // TODO: query for the node outside of the loop
-               // TODO: keep a reference to each node appended and then remove it when the preview is regenerated
-               query("head").append("<link rel=\"stylesheet\" type=\"text/css\" href=\"" + appContext + response.cssMap[media] + "\" media=\"" + media + "\">");
-            }
-         }
-         
-         // Build in the i18n properties into the global object...
-         for (var scope in response.i18nMap)
+         if (originalRequestConfig.lastRequestTimestamp === this._lastRequestTimestamp)
          {
-            if (typeof window[response.i18nGlobalObject].messages.scope[scope] === "undefined")
+            for (var media in response.cssMap)
             {
-               // If the scope hasn't already been used then we can just assign it directly...
-               window[response.i18nGlobalObject].messages.scope[scope] = response.i18nMap[scope];
+               if (response.cssMap.hasOwnProperty(media))
+               {
+                  // TODO: query for the node outside of the loop
+                  // TODO: keep a reference to each node appended and then remove it when the preview is regenerated
+                  query("head").append("<link rel=\"stylesheet\" type=\"text/css\" href=\"" + appContext + response.cssMap[media] + "\" media=\"" + media + "\">");
+               }
             }
-            else
+            
+            // Build in the i18n properties into the global object...
+            for (var scope in response.i18nMap)
             {
-               // ...but if the scope already exists, then we need to mixin the new properties...
-               lang.mixin(window[response.i18nGlobalObject].messages.scope[scope], response.i18nMap[scope]);
+               if (typeof window[response.i18nGlobalObject].messages.scope[scope] === "undefined")
```

Comment:
Feels like `window[response.i18nGlobalObject].messages.scope` should be declared in a variable outside the loop to improve readability?